File: workplace/Synchronization/comparison.jl
using PyPlot
using PyCall
animation = pyimport("matplotlib.animation")


img1 = randn(100, 100);
img2 = randn(100, 100);
imgs = [img1, img2]

fig, ax = plt.subplots()
im1 = ax.imshow(imgs[1])

function update(frame)
    ax.set_title("Frame $frame")
    im1.set_data(imgs[frame%2+1])
    return [im1]
end 
ani = animation.FuncAnimation(fig, update, frames=20, interval=1000, blit=true)

ani.save("rotation_animation.gif", writer="imagemagick", fps=30, dpi=300)
# plt.show()

function plt_imagec(
    img1::AbstractArray{<:Real, 2},
    img2::AbstractArray{<:Real, 2};
    nFrames            = 10       ,
    interval           = 50       ,
    title              = ""       ,
    width              = 5        ,
    height             = 5        ,
    vmaxp              = 100      ,
    vminp              = 0        ,
    vmax               = nothing  ,
    vmin               = nothing  ,
    cmap               = "gray"   ,
    fontsize_title     = 10       ,
    color_facecolor    = "#ffffff",
    color_label        = "#000000",
    )
    @assert size(img1) == size(img2) "The size of two images should be the same."
    imgs = [img1, img2]
    nX, nY = size(img1)
    if vmax === nothing || vmin === nothing
        vmax=quantile(reshape([img1; img2], :), vmaxp/100)
        vmin=quantile(reshape([img1; img2], :), vminp/100)
    end

    fig, ax = plt.subplots(nrows=1, ncols=1, figsize=(width/2.53999863, (nY/nX)*height/2.53999863), facecolor=color_facecolor)
    ax.set_title(title, fontsize=fontsize_title, color=color_label)
    im1 = ax.imshow(imgs[1], cmap=cmap, vmin=vmin, vmax=vmax)
    ax.axis("off")
    fig.tight_layout(pad=0, h_pad=0, w_pad=0)
    function update(frame)
        im1.set_data(imgs[frame%2+1])
        return [im1]
    end 
    ani = animation.FuncAnimation(fig, update, frames=nFrames, interval=interval, blit=true)
    return fig, ani
end

File: workplace/Synchronization/comparison.py
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.animation import FuncAnimation

img1 = np.random.randn(100, 100)
img2 = np.random.randn(100, 100)
imgs = [img1, img2]


img_w  = 'E:/pulseq/20241104_ABDL/syn/1p0_Stitched_w.png'
img_wo = 'E:/pulseq/20241104_ABDL/syn/1p0_Stitched_wo.png'
img1 = plt.imread(img_w)
img2 = plt.imread(img_wo)
imgs = [img1, img2]
titles = ['w/ Synchronization ', 'w/o Synchronization']

fig, ax = plt.subplots()
l = ax.set_title(titles[0])
im1 = ax.imshow(img1, vmin=-1, vmax=1, cmap='gray')
ax.axis('off')
def update(frame):
    print(frame)

    l.set_text(titles[frame])
    im1.set_data(imgs[frame])
    return [im1, l]

# 创建动画
ani = FuncAnimation(fig, update, frames=2, interval=500, blit=False, repeat=True, repeat_delay=0)

# # 显示动画
plt.show()